src/app/modules/app/app.js

```diff
@@ -167,7 +169,8 @@
                 //Search through otherLanguages
                 for(var i = 0; i < $scope.otherLanguages.length; i++) {
                     //If the currentLang is in the other list, add it to the top (visible) list
-                    if($scope.otherLanguages[i].isoCode == $scope.currentLang) {
+                    if($scope.otherLanguages[i].isoCode == $scope.currentLang &&
+                       $scope.otherLanguages[i].name == $scope.currentLangName) {
```

Comment:
I wonder whether it would be worth allowing a match ignoring name if $scope.currentLangName is empty (falsy)? Marginally nicer if someone is advanced enough to be creating these URLs by hand (or just conceivably if they have linked to a filtered list already)